web-view/bootstrap/webpack.config.js

```diff
@@ -13,23 +14,99 @@ module.exports = {
+      "web-pdf_document_properties": path.join(__dirname, "./node_modules/pdf.js/web/pdf_document_properties.js"),
+      "web-pdf_find_bar": path.join(__dirname, "./node_modules/pdf.js/web/pdf_find_bar.js"),
+      "web-pdf_layer_viewer": path.join(__dirname, "./node_modules/pdf.js/web/pdf_layer_viewer.js"),
+      "web-pdf_outline_viewer": path.join(__dirname, "./node_modules/pdf.js/web/pdf_outline_viewer.js"),
+      "web-pdf_presentation_mode": path.join(__dirname, "./node_modules/pdf.js/web/pdf_presentation_mode.js"),
+      "web-pdf_sidebar": path.join(__dirname, "./node_modules/pdf.js/web/pdf_sidebar.js"),
+      "web-pdf_thumbnail_viewer": path.join(__dirname, "./node_modules/pdf.js/web/pdf_thumbnail_viewer.js"),
+      "web-preferences": path.join(__dirname, "./node_modules/pdf.js/web/genericcom.js"),
+      "web-print_service": path.join(__dirname, "./node_modules/pdf.js/web/pdf_print_service.js"),
+      "web-secondary_toolbar": path.join(__dirname, "./node_modules/pdf.js/web/secondary_toolbar.js"),
+      "web-toolbar": path.join(__dirname, "./node_modules/pdf.js/web/toolbar.js"),
+    },
     extensions: ['.tsx', '.ts', '.js']
   },
   output: {
     filename: 'main.js',
     path: path.resolve(__dirname, 'build')
   },
   plugins: [
+    new webpack.DefinePlugin({
+      'typeof PDFJSDev': JSON.stringify('object'),
+      PDFJSDev: `({
+        test: function (code) {
+          if (code === "PRODUCTION") {
+            return true;
+          } else if (code === "GENERIC") {
+            return true;
+          } else if (code === "!PRODUCTION || GENERIC"
+            || code === "!PRODUCTION || GENERIC || CHROME"
+            || code === "GENERIC || CHROME"
+            || code === "CHROME || GENERIC"
+            || code === "MOZCENTRAL || GENERIC"
+            || code === "!PRODUCTION || (GENERIC && !LIB)"
+            || code === "GENERIC && !TESTING") {
+            return true;
+          } else if (code === "GENERIC && !SKIP_BABEL") {
+            return false;
+          } else if (code === "LIB"
+            || code === "LIB && TESTING") {
+            return false;
+          } else if (code === "CHROME"
+            || code === "MOZCENTRAL"
+            || code === "MOZCENTRAL || CHROME"
+            || code === "GECKOVIEW") {
+            return false;
+          } else if (code === "TESTING"
+            || code === "!PRODUCTION || TESTING"
+            || code === "TESTING || MOZCENTRAL"
+            || code === "TESTING || LIB") {
+            return false;
+          } else if (code === "COMPONENTS") {
+            return false;
+          }
+          throw new Error("PDFJSDev-test-" + code);
+        },
+        eval: function (code) {
+          if (code === "BUNDLE_VERSION") {
+            return "4.6.82";
+          } else if (code === "DEFAULT_PREFERENCES") {
+            return window.PDFViewerApplicationOptions.getAll(0x80, true);
```

Comment:
we don't copy the static content of `DEFAULT_PREFERENCES`, then should expose `PDFViewerApplicationOptions` very early.